server/src/main/java/org/tctalent/server/service/db/aws/S3ResourceHelper.java

```diff
@@ -34,292 +51,285 @@
+
+    } catch (Exception e) {
+      log.error("Error uploading file to: " + key, e);
+      throw new FileUploadException("Error uploading file to: " + key, e);
     }
+  }
 
-    public void uploadFile(String content,
-                           String key,
-                           String contentType)
-            throws FileUploadException {
-        try {
-            uploadFile(s3Bucket, content, key, contentType);
-
-        } catch (Exception e) {
-            log.error("Error uploading file to: " + key, e);
-            throw new FileUploadException("Error uploading file to: " + key, e);
-        }
-    }
+  public void uploadFile(String key,
+      MultipartFile source)
+      throws FileUploadException {
+    try {
+      uploadFile(s3Bucket, key, source);
 
-    public void uploadFile(String bucket,
-                           String content,
-                           String key,
-                           String contentType)
-            throws FileUploadException {
-        try {
-            byte[] bytes = content.getBytes("UTF-8");
-            InputStream is = new ByteArrayInputStream(bytes);
-            uploadFile(bucket, is, bytes.length, key, contentType);
-        } catch (FileUploadException e) {
-            throw e;
-        } catch (Exception e) {
-            throw new FileUploadException("Error uploading string content to: " + key, e);
-        }
+    } catch (Exception e) {
+      log.error("Error uploading file to: " + key, e);
+      throw new FileUploadException("Error uploading file to: " + key, e);
     }
+  }
 
-    public void uploadFile(String bucket,
-                           InputStream content,
-                           long contentLength,
-                           String key,
-                           String contentType)
-            throws FileUploadException {
-        try {
-            ObjectMetadata objectMetadata = new ObjectMetadata();
-            objectMetadata.setContentLength(contentLength);
-            objectMetadata.setContentType(contentType);
-            Upload upload = transferManager.upload(bucket, key, content, objectMetadata);
-            // blocking wait, though we could let uploads just run in background
-            upload.waitForCompletion();
-
-            log.debug("uploaded string content to {}", key);
-
-        } catch (Exception e) {
-            log.error("Error uploading string content to: " + key, e);
-            throw new FileUploadException("Error uploading string content to: " + key, e);
-        }
-    }
+  public void uploadFile(String bucket,
+      String key,
+      MultipartFile source)
+      throws FileUploadException {
+    try {
+      ObjectMetadata objectMetadata = new ObjectMetadata();
+      objectMetadata.setContentLength(source.getSize());
+      objectMetadata.setContentType(source.getContentType());
+      Upload upload = transferManager.upload(bucket, key, source.getInputStream(), objectMetadata);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *getInputStream*. Currently, there are execution paths that do not contain closure statements, for example, when TransferManager.upload throws an exception. Ensure the object returned by *getInputStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).